File: app/src/main/java/com/example/mvctutorial/designpattern/_08absttractfactory_guifactory/windows/WindowsButton.java
package com.example.mvctutorial.designpattern._08absttractfactory_guifactory.windows;

import com.example.mvctutorial.designpattern._08absttractfactory_guifactory.Button;

public class WindowsButton implements Button {
    @Override
    public void onClick() {
        System.out.println("windows button click");
    }
}

File: app/src/main/java/com/example/mvctutorial/designpattern/_08absttractfactory_guifactory/windows/WindowsButton.kt
package com.example.mvctutorial.designpattern._08absttractfactory_guifactory.windows

import com.example.mvctutorial.designpattern._08absttractfactory_guifactory.Button

class WindowsButton: Button {
    override fun onClick() {
        println("windows button clicked")
    }
}